src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -398,31 +398,26 @@ private void AddCredentialToCache(Uri serverUri, CURLAUTH authAvail, NetworkCred
             }
         }
 
-        private void AddResponseCookies(Uri serverUri, HttpResponseMessage response)
+        private void AddResponseCookies(EasyRequest state, string cookieHeader)
         {
             if (!_useCookie)
             {
                 return;
             }
 
-            if (response.Headers.Contains(HttpKnownHeaderNames.SetCookie))
+            Uri serverUri = state._requestMessage.RequestUri;
```

Comment:
Is `state._requestMessage.RequestUri` the original URI or the new redirected URI? I wasn't sure looking at this code.

But for this to be correct, the `RequestUri` still needs to be the original URI while processing the response cookies. After the response cookies are processed completely, then the `RequestUri` can be updated to the new redirected URI as given in the 3xx HTTP response.